### PR TITLE
Alire.Properties.Licenses: Add support of SPDX expression 

### LINKED ISSUE #191: Use SPDX license identifiers in the index
As suggested in https://github.com/alire-project/alire/pull/41#discussion_r264729939 it would be nice to use a stardardize way to refer to existing licenses. Opening this issue to track investigation/developments on this.

### PR DESCRIPTION
**Pull request by Fabien-Chouteau**
Fixes #191

### CODE REVIEW

File: src/alire/alire-properties-licenses.adb [hunk 1]
```diff
@@ -1,5 +1,84 @@
...
+          when PostgreSQL         => "PostgreSQL",
+          when Unlicense          => "Unlicense",
+          when WTFPL              => "WTFPL",
+          when Zlib               => "Zlib",
+          when GMGPL_2_0          => "GPL-2.0-only WITH GCC-exception-2.0",
+          when GMGPL_3_0          => "GPL-3.0-only WITH GCC-exception-3.1",
+          when Public_Domain | Custom | Unknown =>
+             raise Checked_Error
+               with Errors.Set ("Invalid legacy license conversion for '" &
+                                  Legacy'Img & "'"));
```
Comment (mosteo) [inline src/alire/alire-properties-licenses.adb]: For my edification: does this mean that SPDX does not support custom licenses? (so we lose this capability?)

Comment (Fabien-Chouteau) [reply src/alire/alire-properties-licenses.adb]: Yes SPDX doesn't allow license outside of the list they provide. And that's going to be a problem for the existing index.

I think there are several options:
 - `Custom` is not allowed anymore and the `licenses` must only be used for SPDX licenses.
 - Add an option to accept `custom` in the SPDX library.
 - Add a detection of `custom` in `Alire` to bypass the SPDX library.

Comment (Fabien-Chouteau) [reply src/alire/alire-properties-licenses.adb]: I will try to add a `other-<License-name-here>` support in the SPDX library.

**Comment by mosteo** [comment]
I think the only thing missing here is bumping the `spdx` dependency to `~0.2`, so the self-build succeeds.

**Comment by Fabien-Chouteau** [comment]
@mosteo This is ready for review,

**Review by mosteo — APPROVED** [review]

**Pull request merged**